### PR TITLE
Bump NodeJS to version 14

### LINKED ISSUE #3849: Update Node version support for WCPay
At the moment supported Node version for WCPay is set to `v12`. If a dev uses a node version higher than the one specified in `.nvmrc`, `npm install` fails with the following errors.

Issue reference - p1642565167088100-slack-CGGCLBN58

`Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: No "exports" main defined in /.npm/_cacache/tmp/git-clonexg93xY/node_modules/@babel/helper-compilation-targets/package.json`

```
$ rm -rf node_modules && rm package-lock.json && npm install
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: woocommerce-payments@3.5.0
npm ERR! Found: @woocommerce/e2e-environment@0.2.3
npm ERR! node_modules/@woocommerce/e2e-environment
npm ERR!   dev @woocommerce/e2e-environment@"^0.2.3" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer @woocommerce/e2e-environment@"0.2.2" from @woocommerce/admin-e2e-tests@0.1.2
npm ERR! node_modules/@woocommerce/admin-e2e-tests
npm ERR!   @woocommerce/admin-e2e-tests@"^0.1.2" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See /.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!    /.npm/_logs/2022-01-19T04_05_45_036Z-debug-0.log
```

Ideally the project should support latest Node & NPM versions.

### PR DESCRIPTION
**Pull request by zmaglica**
Fixes #3849

#### Changes proposed in this Pull Request

Current NodeJS version is 12 which is outdated and the purpose of this PR is to bump the NodeJS version to the latest possible.

I tried to bump version to 16, but some dependencies we depend on haven't updated for a while which caused issues on NodeJS 16. There is a plan to update those dependencies, but for now, the node JS 14 is working fine on environments I tested (The latest Ubuntu and Apple Silicon)

#### Testing instructions

* Make sure that all tests are passing
* Make sure that all E2E tests are passing
* Make sure that you can execute npm scripts without errors

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by htdat** [comment]
FYI. Trigger E2E test run: https://github.com/Automattic/woocommerce-payments/actions/runs/2687595072

**Review by htdat — COMMENTED** [review]
Do we need to make changes for package-lock.json? Based on my tests, we might not need these changes. 

Also, is there any specific reason to use `14.18.2`? Why not use the latest v14 version `14.20.0` at the moment?

**Comment by zmaglica** [comment]
> Do we need to make changes for package-lock.json? Based on my tests, we might not need these changes.
> 
> Also, is there any specific reason to use `14.18.2`? Why not use the latest v14 version `14.20.0` at the moment?

I hit `nvm use 14` and this version was installed. Let me check with the latest one.

**Comment by zmaglica** [comment]
E2E tests have been executed for this branch: https://github.com/Automattic/woocommerce-payments/runs/7838322682?check_suite_focus=true

All tests are green.

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-12.22.6
+14.20.0
```
Comment (achyuthajoy) [inline .nvmrc:1]: Should we specify the whole version number here @zmaglica? It think just `14` works fine.

Someone using NVM without the specific version installed, running `nvm install 14` will always install the latest version. What do you think?

**Review by achyuthajoy — CHANGES_REQUESTED** [review]
@zmaglica Left a comment on the node version. Thanks for working on this.

File: .nvmrc (hunk 1 above)
Comment (zmaglica) [reply .nvmrc:1]: Yeah, we can leave 14. The latest version will always be installed (14.20.0) anyway, so I will make a change to my PR.

Comment (achyuthajoy) [reply .nvmrc:1]: Thanks 👍

**Review by achyuthajoy — APPROVED** [review]

**Comment by haszari** [comment]
Thanks for moving this forward @zmaglica 🚀❤️

**Pull request merged**